Interop/ONNX/src/main/java/org/tribuo/interop/onnx/LabelOneVOneTransformer.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.interop.onnx;
+
+import ai.onnxruntime.OnnxJavaType;
+import ai.onnxruntime.OnnxTensor;
+import ai.onnxruntime.OnnxValue;
+import ai.onnxruntime.OrtException;
+import com.oracle.labs.mlrg.olcut.config.PropertyException;
+import org.tribuo.ImmutableOutputInfo;
+import org.tribuo.classification.Label;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Logger;
+
+public final class LabelOneVOneTransformer extends LabelTransformer {
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = Logger.getLogger(LabelTransformer.class.getName());
+
+    /**
+     * Constructs a Label transformer that operates on a one v one output and produces scores via voting.
+     */
+    public LabelOneVOneTransformer() {
+        super(false);
+    }
+
+    @Override
+    public void postConfig() {
+        if (generatesProbabilities) {
+            throw new PropertyException("", "generatesProbabilities", "generatesProbabilities must not be set to true for this class.");
+        }
+    }
+
+    /**
+     * Rationalises the output of an onnx model into a standard format suitable for
+     * downstream work in Tribuo.
+     *
+     * @param inputs       The onnx model output.
+     * @param outputIDInfo The output id mapping.
+     * @return A 2d array of outputs, the first dimension is batch size, the second dimension is the output space.
+     */
+    @Override
+    protected float[][] getBatchPredictions(List<OnnxValue> inputs, ImmutableOutputInfo<Label> outputIDInfo) {
+        try {
+            if (inputs.size() == 1) {
+                // Single OnnxTensor [batchSize][(numOutputs*(numOutputs-1))/2]
+                if (inputs.get(0) instanceof OnnxTensor) {
+                    OnnxTensor outputScores = (OnnxTensor) inputs.get(0);
+                    if (outputScores.getInfo().type == OnnxJavaType.FLOAT) {
+                        long[] shape = outputScores.getInfo().getShape();
+                        if ((shape.length == 2) && (shape[1] == (outputIDInfo.size() * ((long) outputIDInfo.size() - 1) / 2))) {
```

Comment:
Should this have the `|| shape[1] == 2` as the below clause does?

Reply:
Maybe. ONNX Runtime automatically unfolds the output of a SVMClassifier so it has two columns, but in general if you're building `(n * n-1) / 2` classifiers for your `n` class problem you'll end up with 1 output for binary problems, and if ONNX Runtime hasn't special cased it you'll produce a single output tensor of size `[batch_size,1]` which the other logic covers. Neither ONNX Runtime or the ONNX spec are really correct here, so I added some docs to explain that we're in an area prone to change because it's implementation dependent.